File: CSharp/Day01/Program.cs

using Day01;
using AdventUtilities;

Dictionary<string, string> filePaths = new()
{
    ["example1"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day01", "example1.txt"),
    ["input"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day01", "input.txt")
};

ReadOnlySpan<string> lines = File.ReadAllLines(filePaths["input"]);
int part1 = SafeCracker.TimesZero(lines);
int part2 = SafeCracker.TimesZero(lines, partTwo: true);


Console.WriteLine($"Part 1: {part1}");
Console.WriteLine($"Part 2: {part2}");

File: FSharp/Day01/Program.fs
open AdventUtilities
open SafeCracker

let example = InputData().ReadAllLines 1 "example1" |> List.ofArray
let input = InputData().ReadAllLines 1 "input" |> List.ofArray

let part1 = timesZero stoppedAtZero input
let part2 = timesZero zeroesDuringRotation input

printfn "Part 1: %i" part1
printfn "Part 2: %i" part2
